backend/routes/reservation_routes.py

```
@@ -631,15 +692,36 @@ def update_reservation_status(reservation_id):
+                        logger.warning("Status change notification failed", extra=safe_extra_fields({
+                            'reservation_id': reservation_id,
+                            'recipient_email': reservation.email,
+                            'new_status': new_status,
+                            'error_message': email_result.get('message', 'Unknown error'),
+                            'notification_result': 'failed'
+                        }))
                 else:
-                    current_app.logger.warning(f"No email configuration found for admin or no client email for reservation {reservation.id}")
+                    logger.warning("No email configuration or recipient for status notification", extra=safe_extra_fields({
+                        'reservation_id': reservation_id,
+                        'has_admin_config': admin_email_config is not None,
```

Ruff (0.13.1)

Undefined name `admin_email_config`

(F821)